pkg/endpoint/policy.go

```diff
@@ -670,10 +682,13 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 
 	e.BuildMutex.Lock()
 	defer e.BuildMutex.Unlock()
+	e.ProxyCompletions = &Completions{}
+	defer func() { e.ProxyCompletions = nil }()
```

Comment:
This is very fragile as it depends on the above defer to be called afterwards. Can you do one defer function which sets ProxyCompletions to nil and then unlocks BuildMutex so it's clear that there is a dependency?

Reply:
Good point, with one defer the order is more explicit and it becomes a bit more efficient as well.

Reply:
Good point, with one defer the order is more explicit and it becomes a bit more efficient as well.